chia/server/ws_connection.py

```diff
@@ -122,8 +122,16 @@ async def perform_handshake(self, network_id: str, protocol_version: str, server
             if inbound_handshake_msg is None:
                 raise ProtocolError(Err.INVALID_HANDSHAKE)
             inbound_handshake = Handshake.from_bytes(inbound_handshake_msg.data)
-            if ProtocolMessageTypes(inbound_handshake_msg.type) != ProtocolMessageTypes.handshake:
+
+            # Handle case of invalid ProtocolMessageType
+            try:
+                message_type: ProtocolMessageTypes = ProtocolMessageTypes(inbound_handshake_msg.type)
+            except Exception:
                 raise ProtocolError(Err.INVALID_HANDSHAKE)
+            else:
```

Comment:
this else block is redundant. you would save one level of indentation by removing the `else:` and just execute the two following lines unconditionally.

Also, I think it would be useful to capture the actual message type when there's an error. maybe if could be logged as `INFO`. I assume the only case this would throw is if `inbound_handshake_msg.type` is none of the valid message types.

Reply:
Personally, I like the else clause because I find it cleaner to read, but I'm happy to remove it. In this case, there is an existing warning level message printed from another call that will print the message type (specifically process_msg_and_check). Yes, afaik this will only throw if the message type is none of the valid types.